apps/web/components/NameSignupForm/NameSignupForm.tsx

```diff
@@ -66,24 +77,18 @@ export const NameSignupForm = ({ namespace, slice }: NameSignupFormProps) => {
     validate,
     onSubmit: () => {
       setLoading(true)
-      jsonp(
-        slice.signupUrl
-          .replace('{{EMAIL}}', formik.values.email)
-          .replace('{{NAME}}', formik.values.name)
-          .replace('{{TOGGLE}}', formik.values.toggle),
-        {
-          param: 'c',
+      submit({
+        variables: {
+          input: {
+            signupID: slice.id,
+            email: formik.values.email,
+            name: formik.values.name,
+            toggle: formik.values.toggle === 'Yes',
+          },
         },
-        (err: Error, data: any) => {
-          if (data?.msg.includes('already subscribed')) {
-            setMessage({
-              type: 'error',
-              text: n(
-                'formEmailAlreadyRegistered',
-                'Þetta netfang er þegar á skrá.',
-              ),
-            })
-          } else if (data) {
+      })
+        .then((result) => {
+          if (result.data.mailchimpSubscribe.subscribed) {
```

Comment:
Perhaps have an else clause here like you do in the MailingListSignup component submit handle function